support-lib/objc/DJIMarshal+Private.h

```diff
@@ -118,6 +118,27 @@ struct String {
     }
 };
 
+struct WString {
+    using CppType = std::wstring;
+    using ObjcType = NSString*;
+
+    using Boxed = WString;
+
+    static CppType toCpp(ObjcType string) {
+        assert(string);
+        NSStringEncoding encoding = CFStringConvertEncodingToNSStringEncoding(kCFStringEncodingUTF32LE);
+        NSData* data = [string dataUsingEncoding:encoding];
+        return std::wstring((wchar_t*)[data bytes], [data length] / sizeof (wchar_t));
+    }
+
+    static ObjcType fromCpp(const CppType& string) {
+        assert(string.size() <= std::numeric_limits<NSUInteger>::max());
+        return [[NSString alloc] initWithBytes:string.data()
+                                        length:string.size() * sizeof(wchar_t)
+                                      encoding:NSUTF32LittleEndianStringEncoding];
```

Comment:
@konovalov-aleks still looking for a static_assert to check your assumption that wchar_t is 4 bytes, though it strikes me that maybe that assert should go into the generated code which actually uses wstring.  Otherwise the support-lib code would force a compile failure even if wstrings are not in use.